### DISCUSSION TITLE
Problem with determining the data type, Pandas

### DISCUSSION THREAD

**Discussion by Iriomo** [great-expectations/great_expectations#10537] [Support]
_There is a problem with determining the data type, when running jupiter notepad and the location of the csv file in the system._ 

> !pip install great_expectations
> 
> import great_expectations as gx
> import pandas as pd
> from great_expectations.core.batch import BatchRequest, RuntimeBatchRequest
> import shutil
> import os
> 
> if os.path.exists("./context/gx"):
>     shutil.rmtree("./context/gx") 
> 
> context = gx.get_context(mode="file", project_root_dir="./context")
> 
> print(type(context).__name__)
> 
> source_folder = "C:\\Users\\.....\\data"
> data_source_name = "my_filesystem_data_source"
> data_source = context.data_sources.add_pandas(name=data_source_name)
> 
> 
> asset_name = "World Population"
> path_to_data = "C:\\Users\\....\\data\\world_population.csv"
> asset = data_source.add_csv_asset(asset_name, filepath_or_buffer=path_to_data)
> **ValidationError: 1 validation error for _PandasDataAsset
> type
>   field required (type=value_error.missing)**

_I´m prescribing the type:_

> asset_name = "World Population"
> path_to_data = "C:\\Users\\....\\GX\\data\\world_population.csv"
> asset = data_source.add_csv_asset(asset_name, filepath_or_buffer=path_to_data, type="csv")

_With kwargs:_

> batch_definition_name = "World Population 2022"
> batch_definition = asset.add_batch_definition(
>     name=batch_definition_name)
> 
> batch_parameters = {"reader_method": "read_csv"}
> batch = batch_definition.get_batch(batch_parameters=batch_parameters)
> print(batch.head())
> **BuildBatchRequestError: Bad input to build_batch_request: options is not currently supported for this DataAsset and must be None or {}.**

_Without kwargs:_

> batch_definition_name = "World Population 2022"
> batch_definition = asset.add_batch_definition(
>     name=batch_definition_name)
> 
> batch = batch_definition.get_batch()
> print(batch.head())
> **NotImplementedError: One needs to explicitly provide "reader_method" for Pandas DataAsset extensions as temporary work-around, until "type" naming convention and method for obtaining 'reader_method' from it are established.**

Windows 10
Great Expectations 1.1.3
Python  3.12.4 | packaged by Anaconda, Inc.

**Comment by lucasarruda-ciandt** [comment]
open an issue, they don't seem to come here
